shared/hooks/useSearch.ts

```diff
@@ -22,6 +23,18 @@ export const prepareSearchDocuments = (
 // Modifying single documents or replacing documents with a same length array will therefore NOT trigger an update
 const useSearch = (documents: SearchResult[], query: string): SearchResult[] => {
   const [indexing, setIndexing] = useState(false)
+  const [debouncedQuery, setDebouncedQuery] = useState(query)
+
+  useEffect(() => {
+    const debounceQueryTimeout = setTimeout(() => {
+      setDebouncedQuery(debouncedQuery)
```

Comment:
:question: Shouldn't this be set to `query` after the timeout?
```suggestion
      setDebouncedQuery(query)
```

Reply:
yes, you are absolutely right :speak_no_evil: